### PR TITLE
Ingest decision letter workflow, starter, S3 notification

### PR DESCRIPTION
**Pull request by gnott**
A start on issue https://github.com/elifesciences/elife-bot/issues/965, this addresses the first five checkboxes.

To build out the workflow in small bites, this PR defines the workflow, its starter, and starts a workflow execution after an S3 notification comes into the message queue.

The workflow steps right now only includes `PingWorker`; the real activities to this workflow will be added later. The ping should mean, if all is correct here, that it will cause no trouble, and also do very little at this time.

I also started on issue https://github.com/elifesciences/elife-bot/issues/983 here too, where `workflow/helper.py` module contains a function to define a workflow step, using default values where possible. In future all the workflows can be refactored to use this, if the example here is good.

It was informative to look through the old workflow logic a little, where I noticed the `control` value for workflow activity steps is not used at all. I left it in as a possible value to customise, even though we haven't used it yet, and probably never will. We could possibly remove `control` all together as a separate PR, if this simplification is a good idea (more detail you can see on this line https://github.com/elifesciences/elife-bot/blob/develop/workflow/objects.py#L170 `'control data' is hardcoded in the `boto` call, and the value of `control` is not used).

In summary, this PR represents the basics to add a new workflow, ability to start that workflow, and to start a workflow from an S3 notification from a particularly named bucket. The actual effect of this workflow should be a `ping` and nothing more. This will allow adding activities one-by-one to this workflow and we can review and approve them each separately.

### CODE REVIEW

File: queue_workflow_starter.py [hunk 1]
```diff
@@ -103,9 +108,11 @@ def process_data_pubmedarticledeposit(workflow_data):
...
 
 def process_data_ingestdigest(workflow_data):
-    data = {'info': S3NotificationInfo.from_dict(workflow_data),
-            'run': str(uuid.uuid4())}
-    return data
+    return process_data_s3_notification_default(workflow_data)
+
+
+def process_data_ingestdecisionletter(workflow_data):
+    return process_data_s3_notification_default(workflow_data)
```
Comment (giorgiosironi) [inline queue_workflow_starter.py]: good removal of duplication

File: queue_workflow_starter.py [hunk 2]
```diff
@@ -114,7 +121,8 @@ def process_data_ingestdigest(workflow_data):
     'SilentCorrectionsIngest': process_data_initialarticlezip,
     'PostPerfectPublication': process_data_postperfectpublication,
     'PubmedArticleDeposit': process_data_pubmedarticledeposit,
-    'IngestDigest': process_data_ingestdigest
+    'IngestDigest': process_data_ingestdigest,
+    'IngestDecisionLetter': process_data_ingestdecisionletter
```
Comment (giorgiosironi) [inline queue_workflow_starter.py]: I think you can add a `,` at the end of this line to make diffs shorter in the future

File: starter/starter_IngestDecisionLetter.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import os
+# Add parent directory for imports
+parentdir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```
Comment (giorgiosironi) [inline starter/starter_IngestDecisionLetter.py]: this is weird; is it required if execute `venv/bin/python starter/starter_IngestDecisionLetter.py`?

File: starter/starter_IngestDecisionLetter.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    import settings as settingsLib
+    SETTINGS = settingsLib.get_settings(ENV)
+
+    STARTER_OBJECT = starter_IngestIngestDecisionLetter()
+
+    # note: this starter must be started by an S3Notification and not directly from command line
+    RUN = None
+    INFO = None
+    STARTER_OBJECT.start(settings=SETTINGS, run=RUN, info=INFO)
```
Comment (giorgiosironi) [inline starter/starter_IngestDecisionLetter.py]: there's a lot of boilerplate in here so surprised there is nothing to extract that is shared with other starters?

File: queue_workflow_starter.py (hunk 2 above)
Comment (gnott) [reply queue_workflow_starter.py]: I've noticed having a `,` at the end of a list doesn't harm anything, and it's also not required, so I wasn't sure what is the preferred style. I like adding the `,` as you suggest for better diffs and it works for every line.

File: starter/starter_IngestDecisionLetter.py (hunk 3 above)
Comment (gnott) [reply starter/starter_IngestDecisionLetter.py]: Last time I tried to remove these it resulted in an error invoking the starter. I cannot remember if it was due to a missing `__init__.py` file from the root folder, how the worker runs, or something to do with the NewRelic wrapper, or how the service daemon was invoked.

This hack is present on most, if not all, of the starters right now. 

I suggest if we want to try out removing them again, we could try it on a separate PR, which we can merge and test through to the `end2end` testing to be sure. If it fails there we can revoke the changes and/or diagnose the exact reason why it fails (if it still does). We could create a new issue describing the plan to test / debug / revoke. Do you think this is a good idea @giorgiosironi ?

File: starter/starter_IngestDecisionLetter.py (hunk 4 above)
Comment (gnott) [reply starter/starter_IngestDecisionLetter.py]: Starters were at one time going to be reviewed and refactored. They are very similar but not all the same. I think some starter helpers were added recently. Also, some refactoring of the `if __name__ == "__main__":` parts I did so some in order to get higher test coverage. 

These I think also might be good to be described in a separate issue and completed as part of a separate PR. I thought I had an issue for this (which would have been from years ago) but it may have been closed or it was not exactly this particular topic of refactor.

**Comment by gnott** [comment]
Ah, I see you approved this PR - thanks for reviewing @giorgiosironi! I will create two issues for the things I think could be done in separate PRs so those suggestions are carried forward and the momentum of continuous improvement can go on.

File: starter/starter_IngestDecisionLetter.py (hunk 3 above)
Comment (gnott) [reply starter/starter_IngestDecisionLetter.py]: https://github.com/elifesciences/elife-bot/issues/991 issue I created for this

File: starter/starter_IngestDecisionLetter.py (hunk 4 above)
Comment (gnott) [reply starter/starter_IngestDecisionLetter.py]: https://github.com/elifesciences/elife-bot/issues/992 issue I created for starter refactoring

**Comment by gnott** [comment]
I added `sections.docx` to the continuumtest bucket for these files and it resulted in a completed `IngestDecisionLetter` workflow, where I conclude the code here is functioning correctly.

File: starter/starter_IngestDecisionLetter.py (hunk 3 above)
Comment (giorgiosironi) [reply starter/starter_IngestDecisionLetter.py]: Definitely an issue for a different PR, try to remove from the least important `starter` first.
